bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/discovery/BusGatewayUpnpDiscovery.java

```diff
@@ -0,0 +1,217 @@
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.upnp.UpnpDiscoveryParticipant;
+import org.eclipse.smarthome.config.discovery.upnp.internal.UpnpDiscoveryService;
+//import org.eclipse.smarthome.config.discovery.UpnpDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.jupnp.model.meta.DeviceDetails;
+import org.jupnp.model.meta.ManufacturerDetails;
+import org.jupnp.model.meta.ModelDetails;
+import org.jupnp.model.meta.RemoteDevice;
+import org.jupnp.model.meta.RemoteDeviceIdentity;
+import org.jupnp.model.types.UDN;
+import org.openhab.binding.openwebnet.OpenWebNetBindingConstants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BusGatewayUpnpDiscovery} is responsible for discovering supported BTicino BUS
+ * gateways devices using UPnP. It uses the central {@link UpnpDiscoveryService} implementing
+ * {@link UpnpDiscoveryParticipant}.
+ *
+ * @author Massimo Valla - Initial contribution
+ */
+
+@Component(service = UpnpDiscoveryParticipant.class, immediate = true)
+public class BusGatewayUpnpDiscovery implements UpnpDiscoveryParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(BusGatewayUpnpDiscovery.class);
+
+    public enum BusGatewayId {
+        MH201("IPscenarioModule", "MH201"),
+        MH202("scheduler", "MH202"),
+        F454("webserver", "F454"),
+        MY_HOME_SERVER1("myhomeserver1", "MYHOMESERVER1"),
+        TOUCH_SCREEN_10("ts10", "TOUCHSCREEN10"),
+        MH200N("lightingcontrolunit", "MH200N");
+
+        private final String value, thingId;
+
+        private BusGatewayId(String value, String thingId) {
+            this.value = value;
+            this.thingId = thingId;
+        }
+
+        public static BusGatewayId fromValue(String s) {
+            Optional<BusGatewayId> m = Arrays.stream(values()).filter(val -> s.equals(val.value)).findFirst();
+            return m.orElse(null);
+        }
+
+        public String getThingId() {
+            return thingId;
+        }
+    }
+
+    /**
+     * DeviceInfo bean to store device useful info (and log them)
+     */
+    public class DeviceInfo {
+        private String friendlyName = "<unknown>";
+        private String modelName = "<unknown>";
+        private String modelDescription = "<unknown>";
+        private String modelNumber = "<unknown>";
+        private String serialNumber = "<unknown>";
+        private String host;
+        private String manufacturer = "<unknown>";
+        private UDN udn;
+        private boolean isBTicino = false;
+
+        private DeviceInfo(RemoteDevice device) {
+            logger.debug("+=== UPnP =========================================");
```

Comment:
Please remove log lines with only text. Logging runs in parallel. So in a user system with lots of things these lines just appear randomly between other log messages and thus make them less useful.

Reply:
I grouped the device discovery print in one single line, and removed other multiple lines logs